### PR TITLE
Fix floats with negative exponent

### LINKED ISSUE #8: 5e-3
```glsl
float test = 5.5e-3;
```
is badly parsed as:
```js
[ { type: 'keyword', data: 'float', position: 0, line: 1, column: 5 }
  { type: 'whitespace', data: ' ', position: 5, line: 1, column: 6 },
  { type: 'ident', data: 'test', position: 6, line: 1, column: 10 },
  { type: 'whitespace',
    data: ' ',
    position: 10,
    line: 1,
    column: 11 },
  { type: 'operator', data: '=', position: 11, line: 1, column: 12 },
  { type: 'whitespace',
    data: ' ',
    position: 12,
    line: 1,
    column: 13 },
  { type: 'float', data: '5e', position: 13, line: 1, column: 15 },
  { type: 'operator', data: '-', position: 15, line: 1, column: 16 },
  { type: 'integer', data: '3', position: 16, line: 1, column: 17 },
  { type: 'eof', data: '(eof)', position: 16, line: 1, column: 17 } ]
```
instead of a single literal token

### PR DESCRIPTION
**Pull request by dfcreative**
Closes #8 
@chrisdickinson @mattdesl  please have a look

### CODE REVIEW

**Comment by mattdesl** [comment]
Thanks, looks good to me. @gre ?

File: index.js [hunk 1]
```diff
@@ -314,11 +314,18 @@ function tokenize() {
       return i + 1
     }
 
+    if (c === '-' && /[eE]/.test(last)) {
+      content.push(c);
+      last = c;
+      return i + 1;
```
Comment (gre) [inline index.js]: Can you switch to not using `;`  as it's the code convention style of this project

**Comment by gre** [comment]
Looks good to me, i've tested it a bit, works fine.
Apart from my codestyle comment we are good to merge I think.

**Comment by dfcreative** [comment]
@gre ok

**Comment by mattdesl** [comment]
@gre Added you as a collaborator on the npm module so you can now cut new releases. :smile:

**Comment by gre** [comment]
thanks done :)

**Pull request merged**